### ISSUE TITLE
Can't edit carrier if the ranges are overlapping.

### ISSUE THREAD

**Issue by tautuxs** [PrestaShop/PrestaShop#35674]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you add a carrier and add second range for shipping and save it, it is impossible to edit this carrier. When you go to editing carrier, it stops on the first step "General settings" and the "Next" button is not active, and when you try to click next step "Shipping locations and costs" it stays on the first step showing error "Ranges are overlapping" at the same time not allowing you to change those overlapping ranges.
![image](https://github.com/PrestaShop/PrestaShop/assets/77743151/3fa5a008-1709-499f-9634-1232ff92395f)

### Expected behavior

It should let me press the button "Next" to edit the "Shipping locations and costs".

### Steps to reproduce

1. Go to Shipping > Carriers > Add New Carrier
2. Add name, and transit time and go to next step "Shipipng locations and costs".
3. Set "billing" to weight. 
4. In the field "will be applied when the weight is <" type 10000
5. Select one of the areas (eg. Europe) and type 20
6. Click the button "Add new range" (it should automatically put 10000 in the "Will be applied when the weight is >=" field of the new range).
7. Click "Finish"
8. On the list of carriers click on the newly created carrier (to edit it).
9. Go to the second tab "Shipping locations and costs" - it is impossible, prestashop doesn't allow you to fix the error it displays "Ranges are overlapping".

### PrestaShop version(s) where the bug happened

1.7.8.6

### PHP version(s) where the bug happened

8.0.28

### If your bug is related to a module, specify its name and its version

venipakcarrier v1.0.3

**Comment by tautuxs** [comment]
I edited both ps_range_weight and ps_range_price in my phpmyadmin.
Found 1 delimiter1 that was higher than delimiter2, changed it but that didn't help

**Comment by tautuxs** [comment]
Apparently in the phpMyAdmin database there were 2 lines at **ps_range_weight** with the same **id_carrier** and that's why it was showing this error.

**Comment by paulnoelcholot** [comment]
Hello @Tautvydas3

I reproduce the issue with PrestaShop version 8.1.x, I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

![image](https://github.com/PrestaShop/PrestaShop/assets/112863083/a9c12e1e-83c5-4f10-ac35-fc8a25a78585)

**Comment by Hlavtox** [comment]
Will be fixed by carrier page migration, can you check @jolelievre @boherm?

**Comment by jolelievre** [comment]
Yep I think it's fixed in new page, the form doesn't even let you save overlapping ranges, the CQRS command also triggers an exception And even if it did somehow manage to save or if the data was inserted in another way the tabs are always accessible and not blocked

**Comment by florine2623** [comment]
Hello,

I tried to reproduce this issue with the new carriers page and it is not possible anymore.

<img width="696" alt="Image" src="https://github.com/user-attachments/assets/b112d7e9-5d28-402f-8ed0-33c93a4d3a1d" />

You need to make sure the ranges are not overlapping before saving it. When I try to edit the carrier again, I have no issues either.

This will be fixed in PS 9.0.0.
